File: Examples/Perl/generating-saving/without-cloud-storage/create-barcode-and-get-image-as-stream.pl
use lib 'lib';
use strict;
use warnings;
use utf8;
use File::Slurp; # From CPAN
use JSON;

use AsposeBarCodeCloud::BarcodeApi;
use AsposeBarCodeCloud::ApiClient;
use AsposeBarCodeCloud::Configuration;

    
my $configFile = '../../data/config.json';
my $configPropsText = read_file($configFile);
my $configProps = decode_json($configPropsText);

my $data_path = '../../data/';
my $out_path = $configProps->{'out_folder'};

#ExStart:1
$AsposeBarCodeCloud::Configuration::app_sid = $configProps->{'app_sid'};
$AsposeBarCodeCloud::Configuration::api_key = $configProps->{'api_key'};
$AsposeBarCodeCloud::Configuration::debug = 1;

#Instantiate Aspose.BarCode API SDK
my $barcodeApi = AsposeBarCodeCloud::BarcodeApi->new();

#Set Filename of image
my $name = 'sample-barcode';

#Set Text to encode inside barcode
my $text = 'Aspose.BarCode for Cloud';

#Set Barcode Symbology
my $type = 'datamatrix';

#Set Barcode Image Format
my $format = 'png';

#invoke Aspose.BarCode Cloud SDK API to create barcode and save image to a stream           
my $response = $barcodeApi->GetBarcodeGenerate(text => $text, type => $type, format => $format);

if($response->{'Status'} eq 'OK'){
	#download generated barcode from api response 
	my $output_file = $out_path . $name . '.' . $format;	
	write_file($output_file, { binmode => ":raw" }, $response->{'Content'});
}

#ExEnd:1

File: Examples/Ruby/generating-saving/without-cloud-storage/create-barcode-and-get-image-as-stream.rb
# Require asposecloudsdk

require 'aspose_barcode_cloud'

include AsposeStorageCloud
include AsposeBarcodeCloud

configFile = File.expand_path("../../../data/config.json", __FILE__);
configProps = JSON.parse( IO.read(configFile, encoding: 'utf-8') )
data_path = File.expand_path("../../../data/", __FILE__);
out_path = configProps['out_folder'];

#ExStart:1
AsposeApp.app_key_and_sid(configProps['api_key'], configProps['app_sid'])
AsposeBarcodeCloud::Configuration.debugging = false
AsposeBarcodeCloud::Configuration.temp_folder_path = out_path
AsposeStorageCloud::Configuration.temp_folder_path = out_path

#Instantiate Aspose.BarCode API SDK
@barcodeApi = BarcodeApi.new

#Set Filename of image
name = 'sample-barcode';

#Set Text to encode inside barcode
text = 'Aspose.BarCode for Cloud';

#Set Barcode Symbology
type = 'datamatrix';

#Set Barcode Image Format
format = 'png';

opts = {text: text, type: type, format: format}

#invoke Aspose.BarCode Cloud SDK API to create barcode and save image to a stream
response = @barcodeApi.get_barcode_generate(opts)
response.close
File.rename response.path, out_path + name + '.' + format
print "Requested Barcode has been generated at " + out_path + name + '.' + format
#ExEnd:1
